library/windows/BUILD.gn

```diff
@@ -0,0 +1,68 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build/packaging.gni")
+import("//library/engine.gni")
+import("//library/windows/glfw.gni")
+import("//library/windows/jsoncpp.gni")
+
+action("fetch_glfw") {
+  script = "//tools/dart_tools/bin/fetch_glfw.dart"
+  outputs = glfw_files
+  args = [
+    rebase_path(glfw_download_dir, root_build_dir),
+  ]
+}
+
+action("build_jsoncpp") {
```

Comment:
I'd much prefer that `build_jsoncpp.dart` be broken into two separate scripts, one that downloads the source, and the other that does the build, for a few reasons:
- Having the build done outside of GN is very much against the design of GN, so I'd like to fix that soon (by adding a local GN rule that can build it from source); decoupled scripts means it's easy to swap out just the build portion for GN, without changing anything for the VS build.
- It gives greater flexibility to change the way the source is managed (e.g., moving it out of the build and making it part of some larger pre-build dependency-fetching script, which I'm considering doing soon a short-term step to address the existing issue that downloading all the Windows dependencies at build time is an unusual and potentially surprising setup).
- Even without moving the build into GN, the build step could easily be made much more robust by having a full list of the library sources and headers as a variable that is the output of fetch_jsoncpp and an input to a separate build_jsoncpp, so that if someone modifies jsoncpp locally it will actually rebuild.